From 4698f12a7ac5209c6e25e3d97f15fe803fdf585c Mon Sep 17 00:00:00 2001
From: Shehzan Mohammed <shehzan@arrayfire.com>
Date: Tue, 8 Sep 2015 17:19:36 -0400
Subject: [PATCH] BUGFIX in assign

---
 src/api/c/assign.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/api/c/assign.cpp b/src/api/c/assign.cpp
index 3c93ca2..c990889 100644
--- a/src/api/c/assign.cpp
+++ b/src/api/c/assign.cpp
@@ -231,7 +231,7 @@ af_err af_assign_gen(af_array *out,
         ARG_ASSERT(2, (ndims == 1) || (ndims == (dim_t)lInfo.ndims()));
 
         if (ndims == 1 && ndims != (dim_t)lInfo.ndims()) {
-            af_array tmp_in, tmp_out;
+            af_array tmp_in = 0, tmp_out = 0;
             AF_CHECK(af_flat(&tmp_in, lhs));
             AF_CHECK(af_assign_gen(&tmp_out, tmp_in, ndims, indexs, rhs_));
             AF_CHECK(af_moddims(out, tmp_out, lInfo.ndims(), lInfo.dims().get()));
-- 
2.5.1

